lib/zoneinfo_module.c

```diff
@@ -0,0 +1,2194 @@
+#include <stddef.h>
+#include <stdint.h>
+
+#include "Python.h"
+#include "datetime.h"
+
+// Imports
+PyObject *io_open = NULL;
+PyObject *_tzpath_find_tzfile = NULL;
+PyObject *_common_mod = NULL;
+
+typedef struct _TransitionRuleType _TransitionRuleType;
```

Comment:
You're right about the naming convention - I knew there was something "reserved" about that, but I couldn't remember what it was and had it in the back of my mind to look up. Thanks for pointing it out. Since this whole module is not being exposed in any headers anyway, I think it's safe to just rename these to drop the underscores (maybe I should do that elsewhere to be consistent as well..).

With regards to this line, it is my attempt at a forward declaration, because I want to use `TransitionRuleType` in the `PyZoneInfo_ZoneInfo` declaration, but I don't want to define it until later. If you know of a better way to do the forward declaration I'm happy to change it.